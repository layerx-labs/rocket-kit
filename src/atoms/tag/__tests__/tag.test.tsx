import React from 'react';
import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { Tag } from '../../..';

describe('Tag', () => {
  it('renders', () => {
    const { asFragment } = render(<Tag color={'red'} value={'hello'} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const tagValue = 'This is my tag';
    render(<Tag color={'red'} value={tagValue} />);
    screen.getByText(tagValue);
  });
});
