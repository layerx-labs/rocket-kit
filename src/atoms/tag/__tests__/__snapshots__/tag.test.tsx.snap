// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders 1`] = `
<DocumentFragment>
  .c0 {
  --bg: #ef5766;
  --txt: #ffffff;
  display: inline-block;
  border: 0.125rem solid var(--bg);
  border-radius: 0.25rem;
  background-color: var(--bg);
  padding: 0.1875rem 0.5rem;
  font-size: 0.75rem;
  font-weight: 500;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  line-height: 1;
  color: var(--txt);
  white-space: nowrap;
}

<span
    class="c0 tag"
    color="red"
    title="hello"
  >
    hello
  </span>
</DocumentFragment>
`;
